test/applyfilter.py

```diff
@@ -52,31 +55,64 @@
 import re
 import argparse
 
-# Apply filter on src and create a normalized file dest
-def ApplyOne(src, dest):
+# Apply filter on src and create a normalized file dest.
+# And extract out function summary if specified.
+def ApplyOne(src, dest, summary=None):
     re_addr = re.compile(r'i64 \d{10}\d*')
     re_type = re.compile(r'%("?)(.*?)\.\d+\1 addrspace')
     re_phi = re.compile(r'\[ \d{10}\d*, %')
-    with open(src, 'r') as ins, open(dest, 'w') as outs:
-        for line in ins:
-            line = re_addr.sub(r'i64 NORMALIZED_ADDRESS', line)
-            line = re_type.sub(r'%\1\2.NORMALIZED_TYPEID\1 addrspace', line)
-            line = re_phi.sub(r'[ NORMALIZED_ADDRESS, %', line)
-            outs.write(line)
+    if summary is None:
+        with open(src, 'r') as ins, open(dest, 'w') as outs:
+            for line in ins:
+                line = re_addr.sub(r'i64 NORMALIZED_ADDRESS', line)
+                line = re_type.sub(r'%\1\2.NORMALIZED_TYPEID\1 addrspace', line)
+                line = re_phi.sub(r'[ NORMALIZED_ADDRESS, %', line)
+                outs.write(line)
+    else:
+        re_read_failure = re.compile(r'Failed to read ')
+        re_read_success = re.compile(r'Successfully read ')
+        re_vfcn_failure = re.compile(r' but failed verification')
```

Comment:
This is no longer needed. Verification failures are now asserts (or normal failures in release builds) so they don't need special treatment here.